_posts/2020-09-06-how-the-system-tests-works.md

```diff
@@ -0,0 +1,596 @@
+can be found [here](https://github.com/strimzi/client-examples).
+These type of client we are using as Job. For instance if we want to send some messages we write it like this:
+
+```java
+KubernetesResource.deployNewJob(new JobBuilder()
+    .withNewMetadata()
+        .withNamespace(ResourceManager.kubeClient().getNamespace())
+        .withLabels(producerLabels)
+        .withName(producerName)
+    .endMetadata()
+    .withNewSpec()
+        .withNewTemplate()
+            .withNewMetadata()
+                .withLabels(producerLabels)
+            .endMetadata()
+            .withNewSpec()
+                .withRestartPolicy("OnFailure")
+                .withContainers()
+                    .addNewContainer()
+                    .withName(producerName)
+                        .withImage("strimzi/hello-world-producer:latest")
+                            .addNewEnv()
+                                .withName("BOOTSTRAP_SERVERS")
+                                .withValue(bootstrapServer)
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("TOPIC")
+                                .withValue(topicName)
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("DELAY_MS")
+                                .withValue("1000")
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("LOG_LEVEL")
+                                .withValue("DEBUG")
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("MESSAGE_COUNT")
+                                .withValue(String.valueOf(messageCount))
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("PRODUCER_ACKS")
+                                .withValue("all")
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("ADDITIONAL_CONFIG")
+                                .withValue(additionalConfig)
+                            .endEnv()
+                    .endContainer()
+                .endSpec()
+            .endTemplate()
+        .endSpec()
+        .build());
+```
+
+##### External using Vert.x client
+
+Lastly, we have our external clients implemented by support of Vertx.io Verticles. This type of client has the most limitation such as
+dependent on external listeners, which caused `platform dependency`. 
+Implemented by Abstract recursive builder, which makes these clients very handful. 
+Moreover, we have `Oauth`, `Tracing` and also `Basic` client. Each of them are for specific use case. 
+However, we prefer to always use the internal clients because of that platform dependency. 
+Client can be used for instance:
```

Comment:
Maybe add note that external clients are running where tests are running, and internal inside the kube cluster